### PR TITLE
Flag Observer Mode

### PR DESCRIPTION
**Pull request by killer2k1**
- Adds new "Flag" observer mode.
- [Free Mode] --> [3rd Person Mode] --> [1st Person Mode] --> [Flag Mode]
- Left click toggles back and forth between each flag in Flag Mode.

### CODE REVIEW

File: base/scripts/objectives.cs [hunk 1]
```diff
@@ -1013,17 +1056,22 @@ function checkObjectives(%this)
...
-          
+
           $freeze::FlagClient[%playerTeam] = 0;
           $freeze::FlagClient[%enemyTeam] = 0;
           
           $freeze::OOB[%playerTeam] = false;
           $freeze::OOB[%enemyTeam] = false;
+          
+          Observer::CheckFlagStatus(0);
+          Observer::CheckFlagStatus(1);
```
Comment (dwayneyuen) [inline base/scripts/objectives.cs]: technically we only need to check the flag that was captured right? could we do that to save some cycles?

File: base/scripts/objectives.cs [hunk 2]
```diff
@@ -1013,17 +1056,22 @@ function checkObjectives(%this)
...
 
           Item::hide(%flag, false);
           
+          GameBase::startFadeOut(%flag);
```
Comment (dwayneyuen) [inline base/scripts/objectives.cs:1070]: does this do anything? the other fade outs are started 2.5 seconds before the timers run out - at this point the flag is already gone

File: base/scripts/code.observer.cs [hunk 3]
```diff
@@ -1,231 +1,335 @@
...
+    
+            if($freeze::FlagClient[%flagTeam] == 0) {
+            
+                Observer::setTargetObject($ObserverClient[%i], $teamFlag[%flagTeam]);
+        
+            }
+            else {
+            
+                Observer::setTargetObject($ObserverClient[%i], $freeze::FlagClient[%flagTeam]);
+            }
```
Comment (dwayneyuen) [inline base/scripts/code.observer.cs:91]: seeking to learn - if the flag is being carried what's the difference between setting the observer target object to the client of the flag carrier versus the flag itself?

**Review by dwayneyuen — COMMENTED** [review]
pretty cool - minor comments

File: base/scripts/objectives.cs (hunk 1 above)
Comment (killer2k1) [reply base/scripts/objectives.cs]: correct.

File: base/scripts/objectives.cs (hunk 2 above)
Comment (killer2k1) [reply base/scripts/objectives.cs:1070]: this is more of a viewing visual pleasure when obs'ing the flag after a capture.

File: base/scripts/code.observer.cs (hunk 3 above)
Comment (killer2k1) [reply base/scripts/code.observer.cs:91]: if the flag is being carried you cannot target the flag, if you try this is spawns a visible (to only the observer) flag at the flag stand and continues to bounce up in the air and then fall back down over and over.

**Review by dwayneyuen — APPROVED** [review]

**Pull request merged**